xbmc/utils/EGLUtils.cpp

```diff
@@ -37,3 +43,227 @@ void CEGLUtils::LogError(const std::string& what)
 {
   CLog::Log(LOGERROR, "%s (EGL error %d)", what.c_str(), eglGetError());
 }
+
+CEGLContext::CEGLContext() :
+  m_eglDisplay(EGL_NO_DISPLAY),
+  m_eglSurface(EGL_NO_SURFACE),
+  m_eglContext(EGL_NO_CONTEXT),
+  m_eglConfig(0)
+{
+}
+
+CEGLContext::~CEGLContext()
+{
+  Destroy();
+}
+
+bool CEGLContext::CreateDisplay(EGLDisplay display,
+                                  EGLint renderable_type,
+                                  EGLint rendering_api)
+{
+  EGLint neglconfigs = 0;
+  int major, minor;
+
+  EGLint surface_type = EGL_WINDOW_BIT;
+  // for the non-trivial dirty region modes, we need the EGL buffer to be preserved across updates
+  if (g_advancedSettings.m_guiAlgorithmDirtyRegions == DIRTYREGION_SOLVER_COST_REDUCTION ||
```

Comment:
This needs to be a bit more configurable. I know that e.g. EGL/wayland does not support EGL_SWAP_BEHAVIOR_PRESERVED_BIT at all.